### PR TITLE
Add khttp instrumentation

### PR DESCRIPTION
**Pull request by iNikem**
It was recently removed from upstream

### CODE REVIEW

File: instrumentation/khttp/src/main/java/com/splunk/opentelemetry/khttp/KHttpInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void methodEnter(
+        @Advice.Argument(value = 0) String method,
+        @Advice.Argument(value = 1) String uri,
+        @Advice.Argument(value = 2, readOnly = false) Map<String, String> headers,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+      Context parentContext = currentContext();
+      if (!KHttpTracer.tracer().shouldStartSpan(parentContext)) {
```
Comment (mateuszrzeszutek) [inline instrumentation/khttp/src/main/java/com/splunk/opentelemetry/khttp/KHttpInstrumentation.java:73]: Nit: static import for `tracer()`?

File: instrumentation/khttp/src/main/java/com/splunk/opentelemetry/khttp/KHttpTracer.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  @Override
+  protected TextMapSetter<Map<String, String>> getSetter() {
+    return SETTER;
+  }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.javaagent.khttp-0.1";
```
Comment (mateuszrzeszutek) [inline instrumentation/khttp/src/main/java/com/splunk/opentelemetry/khttp/KHttpTracer.java:72]: Should we use `com.splunk.khttp-0.1` or something similar?

File: instrumentation/khttp/src/test/groovy/com/splunk/opentelemetry/KHttpClientTest.groovy [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ */
+
+
+import io.opentelemetry.instrumentation.test.AgentTestTrait
+import io.opentelemetry.instrumentation.test.base.HttpClientTest
+import khttp.KHttp
+
+class KHttpClientTest extends HttpClientTest<Void> implements AgentTestTrait {
+
+    @Override
```
Comment (mateuszrzeszutek) [inline instrumentation/khttp/src/test/groovy/com/splunk/opentelemetry/KHttpClientTest.groovy:14]: Shouldn't this file be formatted with a 2-space indent?

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/khttp/src/test/groovy/com/splunk/opentelemetry/KHttpClientTest.groovy (hunk 3 above)
Comment (iNikem) [reply instrumentation/khttp/src/test/groovy/com/splunk/opentelemetry/KHttpClientTest.groovy:14]: Yeah, it seems spotless does not apply to groovy here, as we haven't any before

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**